src/ui/MAVLinkDecoder.cc

```diff
@@ -305,6 +305,239 @@ void MAVLinkDecoder::emitFieldValue(mavlink_message_t* msg, int fieldid, quint64
 
     name = name.prepend(QString("M%1:").arg(msg->sysid));
 
+    // handle dynamic field messages
+    if (msgid == MAVLINK_MSG_ID_ESTIMATOR_STATE
+            || msgid == MAVLINK_MSG_ID_ESTIMATOR_STATE_STD
+            || msgid == MAVLINK_MSG_ID_ESTIMATOR_INNOV
+            || msgid == MAVLINK_MSG_ID_ESTIMATOR_INNOV_STD
+            )
+    {
+        // ignore id and sensor fields
+        if (fieldName == "id" || fieldName == "sensor" || fieldName == "n") {
+            return;
+        }
+        if (msgInfo->fields[fieldid].type == MAVLINK_TYPE_FLOAT && msgInfo->fields[fieldid].array_length > 0)
+        {
+            float* nums = (float*)(m+msgInfo->fields[fieldid].wire_offset);
+            uint8_t* id_array = (uint8_t*)(m+msgInfo->fields[3].wire_offset);
+            uint8_t* sensor_array = (uint8_t*)(m+msgInfo->fields[4].wire_offset);
+            fieldType = QString("float[%1]").arg(msgInfo->fields[fieldid].array_length);
+            for (unsigned int j = 0; j < msgInfo->fields[fieldid].array_length; ++j)
+            {
+                uint8_t id_val = id_array[j];
+                QString id = QString("%1").arg(id_val);
+                switch (id_val) {
+                case MAV_FIELD_UNUSED:
+					// use number
+                    break;
+                case MAV_FIELD_POS_N:
+                    id = "pos_N";
+                    break;
```

Comment:
These switch cases for turning the fields and sensor enums into strings could move into some mavlink util header if that would be more appropriate, it would make it simpler for users and less bloated here.